File: problems/problems_1184/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.distanceBetweenBusStops(*test_input)

    def distanceBetweenBusStops(self, distance: List[int], start: int, destination: int) -> int:
        n = len(distance)
        distance += distance
        clock = 0
        for i in range(start, destination if destination > start else destination + n):
            clock += distance[i]
        clockwise = 0
        for i in range(destination, start if start > destination else start + n):
            clockwise += distance[i]
        return min(clock, clockwise)

File: problems/problems_1184/Solution.java
package problems.problems_1184;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int distanceBetweenBusStops(int[] distance, int start, int destination) {
        int n = distance.length, min = Math.min(start, destination), max = Math.max(start, destination);
        int clockwise = 0, counterclockwise = 0;
        for (int i = min; i < max; i++) {
            clockwise += distance[i];
        }
        for (int i = max; i < n + min; i++) {
            counterclockwise += distance[i % n];
        }
        return Math.min(clockwise, counterclockwise);
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] distance = jsonArrayToIntArray(inputJsonValues[0]);
		int start = Integer.parseInt(inputJsonValues[1]);
		int destination = Integer.parseInt(inputJsonValues[2]);
        return JSON.toJSON(distanceBetweenBusStops(distance, start, destination));
    }
}
